clang/test/CodeGenSYCL/lb_sm_90.cpp

```diff
@@ -0,0 +1,49 @@
+// RUN: %clang_cc1 -internal-isystem %S/Inputs -triple nvptx-unknown-unknown -target-cpu sm_90 -fsycl-is-device -Wno-c++23-extensions -S -emit-llvm %s -o - | FileCheck %s --check-prefix CHECK-IR
+// RUN: %clang_cc1 -internal-isystem %S/Inputs %s -triple nvptx64-nvidia-cuda -target-cpu sm_90 -fsycl-is-device -fsyntax-only -Wno-c++23-extensions -verify

```

Comment:
Diagnostic tests should be in Sema. Can you move the diagnostic CHECK to a Sema test? (Don't combine it with AST test as we try to keep AST and diagnostic tests separate)

Reply:
You mean `SemaSYCL`, right? If so, I've split the test in two and moved.